File: Assignment09/Q5.c
/////////////////////////////////////////////////////////////////////////////////
//
//      Include required header files
//
/////////////////////////////////////////////////////////////////////////////////
#include<stdio.h>

/////////////////////////////////////////////////////////////////////////////////
//
//      Function name:  CountDiff
//      Input:          integer
//      Output:         integer
//      Description:    Accept number from user and return difference between
//                      summation of even digits and summation of odd digits
//      Date:           28/05/2026
//      Author:         Pratik Subhash Mahale
//
/////////////////////////////////////////////////////////////////////////////////
int CountDiff(
                int iNo              // Input number
             )
{
    int iDigit = 0;                 // To store digit
    int iSumEven = 0;               // Sum of even digits
    int iSumOdd = 0;                // Sum of odd digits

    // Convert negative number into positive
    if(iNo < 0)
    {
        iNo = -iNo;
    }

    // Traverse digits
    while(iNo > 0)
    {
        iDigit = iNo % 10;          // Extract last digit

        // Check even or odd digit
        if(iDigit % 2 == 0)
        {
            iSumEven = iSumEven + iDigit;
        }
        else
        {
            iSumOdd = iSumOdd + iDigit;
        }

        iNo = iNo / 10;             // Remove last digit
    }

    // Return difference (even - odd)
    return (iSumEven - iSumOdd);
}

/////////////////////////////////////////////////////////////////////////////////
//
//      Application to test CountDiff function
//
/////////////////////////////////////////////////////////////////////////////////
int main()
{
    int iValue = 0;                 // Input number
    int iRet = 0;                   // Result

    printf("Enter number\n");

    scanf("%d", &iValue);

    // Function call
    iRet = CountDiff(iValue);

    // Output display
    printf("%d\n", iRet);

    return 0;                       // Successful termination
}

/////////////////////////////////////////////////////////////////////////////////
//
//      Input:      2395
//      Output:     -15
//
//      Input:      1018
//      Output:     6
//
//      Input:      8440
//      Output:     16
//
//      Input:      5733
//      Output:     -18
//
/////////////////////////////////////////////////////////////////////////////////

/////////////////////////////////////////////////////////////////////////////////
//
//      Time Complexity Analysis
//
//      Let n be the number of digits in the input number
//
//      The while loop runs once for each digit
//      Each iteration performs constant time operations O(1)
//
//      Therefore,
//      Time Complexity = O(n)
//
//      Space Complexity = O(1)
//
/////////////////////////////////////////////////////////////////////////////////

File: Assignment20/Q5.java

///////////////////////////////////////////////////////////////////
//
//      File Name : Q5.java
//      Project   : Logic Building Assignment 20
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Class Name : SmallestDigitLogic
//      Method Name: findSmallestDigit
//      Input      : Integer
//      Output     : Void
//      Description: Find the smallest digit in
//                   the given number
//      Date       : 08/06/2026
//      Author     : Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

class SmallestDigitLogic
{
    void findSmallestDigit(int num)
    {
        int digit = 0;      // Store extracted digit
        int iMin = 9;       // Assume 9 as smallest digit initially

        // Convert negative number to positive
        if(num < 0)
        {
            num = -num;
        }

        // Extract digits and find smallest digit
        while(num != 0)
        {
            digit = num % 10;

            if(digit < iMin)
            {
                iMin = digit;
            }

            num = num / 10;
        }

        // Display result
        System.out.println("Smallest digit is : " + iMin);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Main Class
//
///////////////////////////////////////////////////////////////////

public class Q5
{
    public static void main(String args[])
    {
        // Create object of SmallestDigitLogic class
        SmallestDigitLogic obj = new SmallestDigitLogic();

        // Function call
        obj.findSmallestDigit(45872);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Input  : 45872
//
//      Output :
//      Smallest digit is : 2
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Input  : 90341
//
//      Output :
//      Smallest digit is : 0
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(d)
//
//      Space Complexity : O(1)
//
//      Where d = Number of digits
//
///////////////////////////////////////////////////////////////////
```
